test/python/circuit/library/test_adders.py

```diff
@@ -233,6 +233,57 @@ def test_plugins_when_do_not_apply(self):
             synth = hls(circuit)
             self.assertEqual(synth.count_ops(), {"ModularAdder": 1})
 
+    def test_default_plugins(self):
+        """Tests covering different branches in the default synthesis plugins.."""
+        with self.subTest(name="HalfAdderTest1"):
+            adder = HalfAdderGate(3)
+            circuit = QuantumCircuit(9)
+            circuit.append(adder, range(7))
+            hls = HighLevelSynthesis()
+            _ = hls(circuit)
```

Comment:
Could you add some assertion here that makes sure HLS ran as expected? It can be something simple, like checking for a gate you expect to exist here 🙂

Reply:
Done in [c39d08e](https://github.com/Qiskit/qiskit/pull/13530/commits/c39d08e0f6f91b1d80ddead3f15078b46219670a).